xio-core/src/main/java/com/xjeffrose/xio/client/ClientConfig.java

```diff
@@ -74,4 +92,78 @@ public static ClientConfig fromConfig(String key, Config config) {
   public static ClientConfig fromConfig(String key) {
     return fromConfig(key, ConfigFactory.load());
   }
+
+  public static ClientConfig.Builder newBuilder(ClientConfig fallbackObject) {
+    return new ClientConfig.Builder(fallbackObject);
+  }
+
+  /**
+   * Used to create a ClientConfig at runtime.
+   *
+   * <p>If a value is not set, it defaults to using the default object's value.
+   */
+  public static class Builder {
+    private ClientConfig fallbackObject;
+    private Map<ChannelOption<Object>, Object> bootstrapOptions;
+    private String name;
+    private TlsConfig tls;
+    private boolean messageLoggerEnabled;
+    private InetSocketAddress local;
+    private InetSocketAddress remote;
+    private IdleTimeoutConfig idleTimeoutConfig;
+
+    private Builder(ClientConfig fallbackObject) {
+      this.fallbackObject = fallbackObject;
+    }
+
+    public Builder setBootstrapOptions(Map<ChannelOption<Object>, Object> bootstrapOptions) {
```

Comment:
If I was writing this from scratch i’d use a google auto builder. Don’t change anything here. Just look into it for next time so you don’t have to create a bunch of boilerplate. Then again. I’m hand crafting code with sed and you’re using intellij so i’m sure this was painless.

Reply:
Oh no, it was painful. I just didn't know about auto builder. I'll look into it

Reply:
looked into it. got close to making it work, but it's not compiling and the output is not showing me the error. I'm out of ideas so moving forward with my hand rolled version